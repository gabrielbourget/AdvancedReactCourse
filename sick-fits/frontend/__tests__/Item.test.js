import ItemComponent from '../components/Item';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

const fakeItem = {
	id: 'abc123',
	title: 'A Cool Item',
	price: 50000, 
	description: 'v v cool',
	image: 'dog.jpg',
	largeImage: 'largeDog.jpg'
};

describe.skip('<Item/>', () =>  {

	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<ItemComponent item={ fakeItem }/>);
	});

	it('renders the image properly', () => {
		const img = wrapper.find('img');
		expect(img.props().src).toBe(fakeItem.image);
		expect(img.props().alt).toBe(fakeItem.title);		
	}) 

	it('renders and displays the title and price properly', () => {
		const PriceTag = wrapper.find('PriceTag');
		expect(PriceTag.children().text()).toBe('$500');
		expect(wrapper.find('Title a').text()).toBe(fakeItem.title);
	});

	it('renders out the buttons properly', () => {
		const buttonList = wrapper.find('.buttonList');
		expect(buttonList.children()).toHaveLength(3);
		expect(buttonList.find('Link')).toHaveLength(1);
		// - Alternate way -> expect(buttonList.find('Link').exists()).toBe(true);
		// - Alternate way -> expect(buttonList.find('Link')).toBeTruthy();
		expect(buttonList.find('AddToCart').exists()).toBe(true);
		expect(buttonList.find('DeleteItem').exists()).toBe(true);
	});
});

describe('<Item/> Snapshot Test', () => {
	it('renders and matches the snapshot', () => {
		const wrapper = shallow(<ItemComponent item={ fakeItem }/>);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
