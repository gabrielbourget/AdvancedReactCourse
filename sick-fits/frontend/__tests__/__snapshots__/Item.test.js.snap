// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> Snapshot Test renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Cool Item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        A Cool Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $500
  </PriceTag>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "abc123",
          },
        }
      }
    >
      <a>
        Edit Item
      </a>
    </Link>
    <AddToCart
      id="abc123"
    />
    <DeleteItem
      id="abc123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
